### PR TITLE
[dot/rpc] implement RPC state_getKeysPaged

### LINKED ISSUE #1300: Implement RPC state_getKeysPaged
## Describe the bug

- While testing polkadot.js apps with gossamer node, we see error `TypeError: this._rpcCore.state.getKeysPaged is not a function`  So, polkadot.js apps is expecting state_getKeysPaged to be implemented.

## Expected Behavior

- There should be no error when trying to call RPC method state_getKeysPaged

## Current Behavior

- Calls to state_getKeysPaged fail.

## Possible Solution

- Implement RPC method state_getKeysPaged

## To Reproduce

1. 
2. 
3. 

## Log output

<details>
  <summary>Log Output</summary>
  
  ```Paste log output here
  paste log output...
  ```
</details> 
</br>

## Specification

- go version:
- gossamer version:
- gossamer commit tag:
- gossamer commit hash:
- operating system:
- additional links:

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes

- Implement RPC method `state_getKeysPaged` and tests.
- See https://github.com/w3f/PSPs/blob/master/PSPs/drafts/psp-6.md#1115-state_getkeyspaged 
-

## Tests

```
go test ./dot/rpc/...
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- closes #1300 related to #1198

### CODE REVIEW

**Review by RyRy79261 — APPROVED** [review]

File: dot/rpc/modules/state.go [hunk 1]
```diff
@@ -233,10 +237,24 @@ func (sm *StateModule) GetChildStorageSize(r *http.Request, req *StateChildStora
...
-func (sm *StateModule) GetKeys(r *http.Request, req *StateStorageKeyRequest, res *StateStorageKeysResponse) error {
-	// TODO implement change storage trie so that block hash parameter works (See issue #834)
-	return nil
+// GetKeysPaged Returns the keys with prefix with pagination support.
+func (sm *StateModule) GetKeysPaged(r *http.Request, req *StateStorageKeyRequest, res *StateStorageKeysResponse) error {
+	keys, err := sm.storageAPI.Keys(req.Block)
+	resCount := uint32(0)
+	for _, k := range keys {
+		fKey := fmt.Sprintf("0x%x", k)
+		if strings.HasPrefix(fKey, req.Prefix) &&
```
Comment (noot) [inline dot/rpc/modules/state.go:246]: the trie has a method `GetKeysWithPrefix`, you could add that to the StorageState and use that

File: dot/state/storage.go [hunk 2]
```diff
@@ -321,6 +321,19 @@ func (s *StorageState) Entries(hash *common.Hash) (map[string][]byte, error) {
...
+func (s *StorageState) GetKeysWithPrefix(hash *common.Hash, prefix []byte) ([][]byte, error) {
+	if hash == nil {
+		sr, err := s.blockState.BestBlockStateRoot()
+		if err != nil {
+			return nil, err
+		}
+		hash = &sr
+	}
+
+	return s.tries[*hash].GetKeysWithPrefix(prefix), nil
```
Comment (noot) [inline dot/state/storage.go:334]: can you error or load trie from database in the case `s.tries[*hash]` is nil?

Comment (edwardmack) [reply dot/state/storage.go]: yes, good catch.  All set.

**Review by noot — APPROVED** [review]
great work!

**Pull request merged**